app/models/comment.rb

```diff
@@ -0,0 +1,30 @@
+class Comment < ActiveRecord::Base
+  belongs_to :article
+  attr_accessible :content, :email, :subject
+  validates :subject, presence: true
+  validates :email, presence: true
+  validates_format_of :email, :with => /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i
+
+  def self.cached_article(id)
+    Rails.cache.fetch(["#{self.name}_article", id]) do
+      find(id).article
+    end
+  end
+
+  def self.cached_find(id)
+    Rails.cache.fetch([self.name, id]) do
+      find(id)
+    end
+  end
+
+  def self.cached_all(article_id)
+    Rails.cache.fetch([self.name, 'all_#{article_id}']) do
```

Comment:
@phendych you don't need to be doing a string interpolation here, it receives an array of strings to build the cache keys, for example you can pass `.fetch([self.name, article_id, 'all'])`